From 04a1ef7a7d4a518e1031ffdf83252a0029477470 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Sun, 13 Dec 2020 19:45:47 -0800
Subject: [PATCH 23/25] SystemUI: Improve volume panel expansion/contraction
 animation

- Always decelerate instead of accelerating for a more responsive feel
  (this shouldn't have anything to do with the expand/contract
  direction)
- Use the respective show/hide animation durations, as 80 ms is too
  fast to look smooth on 60 and 90 Hz displays

Change-Id: I31c2fe3da2a237addbb6052bdf83f53949ee13e3
---
 .../src/com/android/systemui/volume/VolumeDialogImpl.java    | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
index add29ce5..59136386 100644
--- a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
+++ b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogImpl.java
@@ -613,9 +613,8 @@ public class VolumeDialogImpl implements VolumeDialog,
                 }
             }
         });
-        animator.setInterpolator(expand ? new SystemUIInterpolators.LogDecelerateInterpolator()
-                : new SystemUIInterpolators.LogAccelerateInterpolator());
-        animator.setDuration(UPDATE_ANIMATION_DURATION);
+        animator.setInterpolator(new SystemUIInterpolators.LogDecelerateInterpolator());
+        animator.setDuration(expand ? DIALOG_SHOW_ANIMATION_DURATION : DIALOG_HIDE_ANIMATION_DURATION);
         animator.start();
     }
 
-- 
2.29.2

